src/widgetastic_patternfly/__init__.py

```diff
@@ -1804,3 +1805,140 @@ def items(self):
             # value will include the label from the <strong> block, parse it out
             items.update({key: element_text.replace(key, '', 1).lstrip()})
         return items
+
+
+class DatePicker(Widget):
+    """Represents the Bootstrap DatePicker.
+
+      Args:
+        name: name of DatePicker
+        id: id of DatePicker
+
+    .. code-block:: python
+
+       date = DatePicker(name='miq_date_1')
+    """
+    ROOT = ParametrizedLocator(".//*[contains(@class, 'datepicker-dropdown')]")
+    TEXT_BOX = ".//*[@name='{name}' or @id='{id}']"
+    DATE = ".//*[contains(@class, 'datepicker-days')]/table/tbody/tr/*"
+    MONTH = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+    YEAR = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+    PREV = ".//*[contains(@class, 'prev')]"
+    NEXT = ".//*[contains(@class, 'next')]"
+    DATE_PICKER_SWITH = ".//*[contains(@class, 'datepicker-switch')]"
+    DATE_MAPPING = {'dd': '%d', 'mm': '%m', 'MM': '%B', 'yy': '%y', 'yyyy': '%Y'}
+
+    def __init__(self, parent, name=None, id=None, logger=None):
+        """Create the widget"""
+        Widget.__init__(self, parent, logger=logger)
+        self.name = name
+        self.id = id
+        self.TEXT_BOX = self.TEXT_BOX.format(name=name, id=id)
+
+    @property
+    def value(self):
+        return self.parent_browser.get_attribute('value', self.TEXT_BOX)
+
+    @property
+    def date_switch_value(self):
+        return self.browser.get_attribute('textContent', self.DATE_PICKER_SWITH)
+
+    def _select_date(self, dd):
+        for el in self.browser.elements(self.DATE):
+            if (not bool({'old', 'new'} & self.browser.classes(el))) and (int(el.text) == int(dd)):
+                el.click()
+                return True
+
+    def _select_month(self, month):
+        for el in self.browser.elements(self.MONTH):
+            if el.text == month:
+                el.click()
+                return True
+
+    def _select_year(self, year):
+        for yr in range(1, 10):
+            start_yr, end_yr = [int(item) for item in self.date_switch_value.split('-')]
+            if start_yr <= int(year) <= end_yr:
+                for el in self.browser.elements(self.YEAR):
+                    if int(el.text) == int(year):
+                        el.click()
+                        return True
+            elif int(year) < start_yr:
+                self.browser.click(self.PREV)
+            elif int(year) > end_yr:
+                self.browser.click(self.NEXT)
+        else:
+            raise ValueError("Year not valid")
+
+    def read(self):
+        return self.value
+
+    def select(self, date):
```

Comment:
QUESTION: what is it for ? there is fill method.

Reply:
now it's internal method :)